crowbar_framework/app/models/service_object.rb

```diff
@@ -1225,7 +1226,8 @@ def apply_role(role, inst, in_queue)
       pids = {}
       unless non_admin_nodes.empty?
         non_admin_nodes.each do |node|
-          nobj = pre_cached_nodes[node] || NodeObject.find_node_by_name(node)
+          pre_cached_nodes[node] ||= NodeObject.find_node_by_name(node)
```

Comment:
maybe I am a bit blind here, but how can `pre_cached_nodes[node]` ever be not empty at this point?